### PR TITLE
fix: only scroll window if selected suggestion is out of view

### PR DESCRIPTION
**Pull request by rambah**
**What kind of change does this PR introduce?** 

- [X] Bugfix
- [X] Feature
- [ ] Documentation
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [X] No

**The PR fulfills these requirements:**

- [X] It's submitted to the `develop` branch (or `v[X]` branch)
- [X] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix: [#xxx]`, where "xxx" is the issue number)
- [X] Any necessary documentation has been added or updated [in the docs](https://github.com/digitalfortress-tech/localstorage-slim#readme)  or explained in the PR's description.

**Other information:**
Previously, when navigating through a list of suggestions using the arrow keys, the window would scroll to center the selected item every time, even if it was already in view. This update modifies that behavior: the window will now only scroll when necessary to keep the selected item visible. This change enhances the overall usability of the package by providing a smoother and more intuitive browsing experience.

### CODE REVIEW

**Comment by niketpathak** [comment]
Thank you for your contribution, I will test it and get back when able. Thanks again! :)

**Comment by niketpathak** [comment]
After a quick test, this doesn't seem to work as expected. 

Here's a screencast showing the behaviour. 

[Screencast from 21-06-2024 18:13:31.webm](https://github.com/digitalfortress-tech/typeahead-standalone/assets/10226768/8e232cdf-689a-405c-9a70-566a5d4c9f1a)

**Comment by rambah** [comment]
Oh, yeah my fault. Instead of using the scrollBy function of the container, I used that of the window. Now it should be fixed :) See following screencast:

https://github.com/digitalfortress-tech/typeahead-standalone/assets/34490250/77fdef93-f3f6-41cb-ba90-27156139ebec

**Comment by niketpathak** [comment]
Hello, what you do, is exactly similar to 

```ts
listContainer.querySelector(`.${classNames.selected}`)?.scrollIntoView({ block: 'nearest' });
```

I think this single line achieves what is needed.

File: src/typeahead-standalone.ts [hunk 1]
```diff
@@ -377,8 +377,31 @@ const typeahead = <T extends Dictionary>(config: typeaheadConfig<T>): typeaheadR
     // update hint if its enabled
     hint && updateHint(selected || resultSet.hits[0]);
 
+    const scrollIntoViewIfNeeded = (container: HTMLDivElement, element: HTMLDivElement) => {
```
Comment (niketpathak) [inline src/typeahead-standalone.ts:380]: this function is not needed

File: src/typeahead-standalone.ts [hunk 2]
```diff
@@ -377,8 +377,31 @@ const typeahead = <T extends Dictionary>(config: typeaheadConfig<T>): typeaheadR
...
+
+      if (isLeft) {
+        container.scrollBy({ left: elementRect.left - containerRect.left - 10, behavior: 'smooth' });
+      } else if (isRight) {
+        container.scrollBy({ left: elementRect.right - containerRect.right + 10, behavior: 'smooth' });
+      }
+    };
     // scroll when not in view
-    listContainer.querySelector(`.${classNames.selected}`)?.scrollIntoView({ block: 'center' });
+    scrollIntoViewIfNeeded(listContainer, listContainer.querySelector(`.${classNames.selected}`) as HTMLDivElement);
```
Comment (niketpathak) [inline src/typeahead-standalone.ts:404]:
```suggestion
    listContainer.querySelector(`.${classNames.selected}`)?.scrollIntoView({ block: 'nearest' });
```

**Comment by rambah** [comment]
You are absolutely right 😳 Didn't know that was possible with scrollIntoView...

**Comment by rambah** [comment]
I added your suggested line. Should work now :) Thanks for reviewing it.

**Pull request merged**